internal/gatewayapi/route.go

```diff
@@ -736,42 +808,93 @@ func (t *Translator) processGRPCRouteRules(grpcRoute *GRPCRouteContext, parentRe
 		}
 
 		// process each ir route
-		for _, irRoute := range ruleRoutes {
-			irRoute.IsHTTP2 = true
-			destination := &ir.RouteDestination{
-				Settings: allDs,
-				Metadata: buildResourceMetadata(grpcRoute, rule.Name),
-			}
-
-			switch {
-			// If the route already has a direct response or redirect configured, then it was from a filter so skip
-			// processing any destinations for this route.
-			case irRoute.DirectResponse != nil || irRoute.Redirect != nil:
-			// return 500 if any destination setting is invalid
-			// the error is already added to the error list when processing the destination
-			case failedProcessDestination:
+		destination := &ir.RouteDestination{
+			Settings: allDs,
+			Metadata: buildResourceMetadata(grpcRoute, rule.Name),
+		}
+		switch {
+		// return 500 if any destination setting is invalid
+		// the error is already added to the error list when processing the destination
+		case processDestinationError != nil:
```

Comment:
this is a big diff, can we reduce the diff to only capture the logging change ?

Reply:
I made this change to make it easier to collect the `routesWithDirectResponse` for different reasons. Without this change, the code would be a bit fragile and harder to follow.

And this is not a big change if you look into the code - it just moved the switch out and for loop in.